### ISSUE TITLE
Problem referencing nodes from permission queries created during nested mutation operations

### ISSUE THREAD

**Issue by wallslide** [graphcool/api-bugs#103]
## Background
I have a `User` model, an `Event` model, and a `Location` model. A `User` can be the owner of many `Events`. And an `Event` should have a single `Location`.

I want to perform a `createEvent` mutation, and use nested mutations to create the event's connected `Location` node at the same time. 

The problem comes when I want to enforce the following constraint: 
> A User can create a Location if it belongs to an Event that they own

I created the following permission query on the `Location` model to enforce the above constraint:
```
query ($input_eventId: ID!, $user_id: ID!) {
  SomeEventExists(filter:{
    id: $input_eventId,
    owner: {
      id:$user_id
    }
  })
}
```
However, this permission fails. My theory is that because I am using nested mutations, the event that I need the `SomeEventExists` function to find doesn't exist yet due to some race condition. To test this theory, instead of using a nested query, I first performed the `createEvent` mutation, and then using the resulting `eventId`, performed a separate `createLocation` mutation. This time I had no permission problems.

## Summary
In summary, it seems like nested mutations introduce some uncertainty into the permission query system, making it unreliable to enforce constraints across nodes that are being created as part of the nested query operation.

**Comment by marktani** [comment]
Thanks for the detailed write up!

This is not a bug, but rather an unfortunate behaviour. Let's discuss potential approaches here https://github.com/graphcool/feature-requests/issues/246 🙂
